core/rawdb/accessors_chain.go

```diff
@@ -754,29 +754,48 @@ func DeleteReceipts(db ethdb.KeyValueWriter, hash common.Hash, number uint64) {
 	}
 }
 
+func IsGenesisHash(db ethdb.Reader, hash common.Hash) bool {
```

Comment:
This is a fairly expensive function to call every block, IIUC. Would be better to cache genesis hashes for faster searching.

Another optimization might be to fork/join the search over parallel threads, but this would only be useful if you chose not to cache the hashes for some reason

Reply:
I agree that there is an easy optimization exists here. i can create an issue for the future and merge this for now

Reply:
https://github.com/dominant-strategies/go-quai/issues/2097